src/controllers/userController.ts

```diff
@@ -0,0 +1,67 @@
+import { Response, Request } from 'express';
+import MyCustomError from '../utils/customError';
+import { catchAsync } from '../middlewares/error';
+import { userService } from '../services/userService';
+import { userDTO } from '../dto/user.dto';
+
+let certificationNumber: string;
+
+class userController {
+  constructor(private Service: userService) {}
+
+  public checkEmail = catchAsync(async (req: Request, res: Response) => {
+    const email: string = req.body.email;
+
+    if (!email) throw new MyCustomError('이메일을 확인해 주세요.', 400);
+
+    await this.Service.checkEmail(email);
+    return res.status(200).json({ message: '사용 가능한 이메일입니다.' });
+  });
+
+  public sendEmail = catchAsync(async (req: Request, res: Response) => {
+    const email: string = req.body.email;
+    certificationNumber = Math.random().toString().substring(2, 8);
+
+    await this.Service.emailSend(email, certificationNumber);
+    return res.status(200).json({ message: '인증 이메일이 발송 되었습니다.' });
+  });
+
+  public confirmEmail = catchAsync(async (req: Request, res: Response) => {
+    const userEmailCode: string = req.body.userEmailCode;
+
+    if (!userEmailCode) {
+      throw new MyCustomError('인증번호를 확인해 주세요.', 400);
+    }
+
+    if (userEmailCode !== certificationNumber) {
+      throw new MyCustomError('인증번호가 일치하지 않습니다.', 400);
+    }
+
+    return res.status(200).json({ message: '인증이 성공적으로 완료되었습니다.' });
+  });
+
+  public signUp = catchAsync(async (req: Request<{}, {}, userDTO, {}>, res: Response) => {
+    const { email, password, nickname } = req.body;
+
+    if (!email || !password) {
+      throw new MyCustomError('이메일과 비밀번호를 입력해주세요', 400);
+    }
+
+    await this.Service.signUp(email, password, nickname as string);
```

Comment:
```suggestion
    if (!nickname) throw new MyCustomError('닉네임을 입력해주세요', 400);

    await this.Service.signUp(email, password, nickname);
```
타입단언(as)는 지양해야 하며, 대신에 if문을 활용하여 예외처리를 해주면 즉각적으로 오류를 반환하게 됩니다. 이 방식으로 nickname의 타입도 string으로 추론됩니다.